lua/telescope/builtin/git.lua

```diff
@@ -110,10 +110,12 @@ git.status = function(opts)
   local results = {}
   for _, v in ipairs(output) do
     if v ~= "" then
-      local mod, fname = string.match(v, '(..)%s(.+)')
-      if mod ~= 'A ' and mod ~= 'M ' and mod ~= 'R ' and mod ~= 'D ' then
-        table.insert(results, { mod = mod, file = fname })
+      local mod, file = string.match(v, '(..)%s(.+)')
+      local old = ''
+      if mod:sub(1, 1) == 'R' or mod:sub(1, 1) == 'C' then
+        old, file = string.match(file, '(.+)%s.+%s(.+)')
       end
+      table.insert(results, { mod = mod, old = old, file = file })
```

Comment:
We should just move that into entry_maker. So we remove that whole `results` part, and just do `results = output` at line 130. That way you don't have to concat the lines back together you can just do `ordinal = line` and `display = line`.
Does that make sense to you?

Reply:
But you should do a `if entry == '' then return nil end` check inside the entry_maker because we will end up with one empty line and that would be added. You won't see it but it is selectable.

Reply:
I'm totally for that. I asked about the empty result to actually go even further and rely on finders.new_oneshot_job instead of `vim.split(utils.get_os_command_output('git status -s'), '\n')`. But moving everything we can in entry_maker is fine already.

Reply:
But moving to new_oneshot_job will always open the window and pull results async. We can't just not open it with that solution. Because we don't even know if there will be results at this point in this. Also doing this upfront is no real "performance" hit.